docs/source/blueprints/BP003.rst

```diff
@@ -0,0 +1,549 @@
+           first sight, it seems that the task monitoring should use
+           the same repository of status and update it in a similar
+           way, but on behalf of the "lost/exterminated task".
+
+Suggested Terminology for the Task Phases
+-----------------------------------------
+
+Task execution has been requested
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A Task whose execution was requested by the user. All of the tasks on a Job's
+``test_suite`` attribute are requested tasks.
+
+If a software component deals with this type of task, it's advisable that it
+refers to ``TASK_REQUESTED`` or ``requested_tasks`` or a similar name that links
+to this definition.
+
+Task is being triaged
+~~~~~~~~~~~~~~~~~~~~~
+
+The details of the task are being analyzed, including, and most importantly,
+the ability of the system to *attempt* to fulfill its requirements. A task
+leaves triage, and it's either considered "discarded" or proceeds to be prepared
+and then executed.
+
+If a software component deals with this type of task, for instance, if a "task
+scheduler" is looking for runners matching the Task's kind, it should keep it
+under a ``tasks_under_triage`` or mark the tasks as ``UNDER_TRIAGE`` or
+``TRIAGING`` a similar name that links to this definition.
+
+Task is being prepared
+~~~~~~~~~~~~~~~~~~~~~~
+
+Task has left triage, and it has not been discarded, that is, it's a candidate
+to be set up, and if it goes well, executed.
+
+The requirements for a task are being prepared in its respective isolation
+model/execution environment, that is, the spawner it'll be executed with is
+known. The setup actions will be visible by the task.
+
+If a software component deals with this type of task, for instance, the
+implementation of the resolution of specific requirements, it should keep it
+under a ``tasks_preparing`` or mark the tasks as ``PREPARING`` or a similar name
+that links to this definition.
+
+Task is ready to be started
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Task has been prepared successfully, and can now be executed.
+
+If a software component deals with this type of task, it should keep it under a
+``tasks_ready`` or mark the tasks as ``READY`` or a similar name that links to
+this definition.
+
+Task has been started
+~~~~~~~~~~~~~~~~~~~~~
+
+A task was successfully started by a spawner.
+
+Note that it does *not* mean that the test that the task runner (say, an
+"avocado-runner-$kind task-run" command) will run has already started. This will
+be signaled by a "status: started" kind of message.
+
+If a software component deals with this type of task, it should keep it under a
+``tasks_started`` or mark the tasks as ``STARTED`` or a similar name that links
+to this definition.
+
+Task has failed
+~~~~~~~~~~~~~~~
+
+Quite self-explanatory. If the spawner failed to start a task, it should be kept
+under a ``tasks_failed`` structure or be marked as ``FAILED``
+or a similar name that links to this definition.
```

Comment:
IMHO, we should have a state that covers all the failures before a Task goes to `READY` and a state `FAILED`, which includes failures between `READY` and `FINISHED`. It is more user-friendly knowing a problem happened during the Task preparation or the Task execution.

Reply:
> IMHO, we should have a state that covers all the failures before a Task goes to `READY` and a state `FAILED`, which includes failures between `READY` and `FINISHED`. It is more user-friendly knowing a problem happened during the Task preparation or the Task execution.

I think the flow can be simplified by having one and only `FINISHED` phase, because this is really about its state, and not really about the Task (think of the test) result.  So, I'm consolidating all in a single `FINISHED` phase (a real end of the line) and considering that the extra information about why it end up there will be kept.